get-started: add genai guide

craig-osterhout opened the pull request:
### Proposed changes

Add GenAI use-case guide.

Sample app for testing:
https://github.com/craig-osterhout/genai-guide

### Related issues (optional)

ENGDOCS-1768

https://deploy-preview-18944--docsdocker.netlify.app/guides/use-case/genai-pdf-bot/

slimslenderslacks commented:
@craig-osterhout the model-puller Dockerfile takes up a lot of space and we it might be a bit distracting for readers.  Given that we want to keep their focus on the pull, we could moving most of this code into a public image (eg something like docker/pull-ollama-model) and then it's just a simple compose.yaml file change with no requirement on a Dockerfile.

slimslenderslacks commented:
I'm trying to figure out what a docker ai companion notebook would be for this project.

dvdksn commented on content/guides/use-case/genai-pdf-bot/containerize.md:
````diff
@@ -0,0 +1,120 @@
+$ docker compose up --build
+```
+
+Open a browser and view the application at [http://localhost:8000](http://localhost:8000). You should see a simple Streamlit application.
````
Not sure if we want to point out here that starting the service may take a little while depending on your network connection, because it involves downloading quite a few dependencies.

dvdksn commented on content/guides/use-case/genai-pdf-bot/develop.md:
```diff
@@ -0,0 +1,286 @@
+    healthcheck:
+      test: ["CMD-SHELL", "wget --no-verbose --tries=1 --spider localhost:7474 || exit 1"]
+      interval: 5s
+        timeout: 3s
+        retries: 5
```
Indentation issue
```suggestion
      timeout: 3s
      retries: 5
```

craig-osterhout commented:
Applied feedback.
- Updated the sample app so that OpenAI appears as an option in the initial streamlit form.
- Overhauled the Add an LLM service section to include OpenAI and better guide the user on which option to choose.
- Updated the pull-model service to use the pre-built image.
